internal/pkg/bulk/engine.go

```diff
@@ -115,6 +237,30 @@ func (b *Bulker) Client() *elasticsearch.Client {
 	return client
 }
 
+func (b *Bulker) Tracer() *apm.Tracer {
+	return b.tracer
+}
+
+// check if remote output cfg changed
+func (b *Bulker) CheckRemoteOutputChanged(zlog zerolog.Logger, name string, newCfg map[string]interface{}) bool {
+	curCfg := b.remoteOutputConfigMap[name]
+
+	hasChanged := false
+
+	// TODO remoteOutputConfigMap empty when FS restarts - won't detect changes
+	// when output config first added, not reporting change
```

Comment:
In what cases is this causing issues?
From what I can see, `CheckRemoteOutputChanged` is only called by `CreateAndGetBulker`; if fleet-server restarts/starts up and the config map is empty, the bulker will be nil and a new one is created.

But I think that this function should not be public, and should also be changed to show it may mutate the config map as well

Reply:
I think the issue is that after a fleet-server restart, if the remote config changed (e.g. hosts), the code doesn't detect that it changed, and not going to recreate the api key. 
However when I tested this, I think eventually the changes were detected, so might not be an actual issue.

I'm renaming it and make it private.